### ISSUE TITLE
Upgrade on addons does not happen after a failed attempt to bring up addons.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rke#718]
rke version v0.1.8-rc5

Steps to reproduce the problem:
1. Have some invalid yml syntax in addon part of the cluster.yml and attempt rancher up.
2. Add ons fails to start with following error which is expected:
```
INFO[0027] [addons] Executing deploy job..              
ERRO[0089] Failed to get job complete status: <nil>     
WARN[0089] Failed to deploy addon execute job [rke-user-includes-addons]: Failed to get job complete status: <nil> 
INFO[0089] Finished building Kubernetes cluster successfully 
```
3. Correct the addon part of the cluster.yml for any errors and attempt rancher up.

Even though RKE indicates that User addons deployed successfully , addon job is not attempted.

A subsequent rancher up , results in addons being deployed.

**Comment by deniseschannon** [comment]
Available with v0.1.8-rc6

**Comment by soumyalj** [comment]
@moelsayed : Tested with rke version v0.1.8-rc8. I still see this issue.
The cluster.yml has a simple addon which has nginx container .
```
auth:
  strategy: x509
nodes:
  - address: abc
    user: root
    role: [controlplane, etcd, worker]
addons: |-
  apiVersion: v1
  kind: Pods
  metadata:
    name: my-nginx
    namespace: default
  spec:
    containers:
    - name: my-nginx
      image: nginx
      ports:
      - containerPort: 80
```
Addon failed as expected as kind is ```Pods```
After fixing the error, rancher up was executed again and it still fails
```
INFO[0033] [addons] Executing deploy job..              
ERRO[0094] [k8s] Job [rke-user-addon-deploy-job] is not deleted 
WARN[0094] Failed to deploy addon execute job [rke-user-includes-addons]: [k8s] Job [rke-user-addon-deploy-job] is not deleted 
INFO[0094] Finished building Kubernetes cluster successfully 



```

**Comment by moelsayed** [comment]
@soumyalj This is a timing issue. 

Kubernetes doesn't allow job definitions to be updated. So, internally, when an addon is updated, we remove the user-addon-job and redeploy it. This can take too long sometimes. and would timeout as in this case; RKE waited for 60 seconds before giving up. 

We wait 2 x `addon_job_timeout` for this process to finish. So, can you please try again with `addon_job_timeout: 30` and with debug enabled?

**Comment by moelsayed** [comment]
@soumyalj I updated the error message to be more descriptive.

**Comment by soumyalj** [comment]
This will be tested again once the new rc is available .

**Comment by deniseschannon** [comment]
Available in v0.1.8-rc11

**Comment by moelsayed** [comment]
I managed to reproduce the timeout. Here is what I found:

We deploy addon jobs with `backoffLimit` of `10`. Since kubernetes `v1.10.0`, the [backofflimit is broken](https://github.com/kubernetes/kubernetes/issues/62382).

When a broken user addon, is deployed using rke `v0.1.8-rc8` which defaults to k8s `v1.10.3-rancher2-1`, we can see the following:
```
$ kubectl get pod --all-namespaces | grep rke-user-addon-deploy-job

ube-system     rke-user-addon-deploy-job-27p7r           0/1       Error               0          1m
kube-system     rke-user-addon-deploy-job-29pcn           0/1       ContainerCreating   0          2s
kube-system     rke-user-addon-deploy-job-2bslc           0/1       Error               0          24s
kube-system     rke-user-addon-deploy-job-2dpnh           0/1       Error               0          47s
kube-system     rke-user-addon-deploy-job-2r6v8           0/1       Error               0          3m
kube-system     rke-user-addon-deploy-job-44nq2           0/1       Error               0          41s
kube-system     rke-user-addon-deploy-job-4h2r9           0/1       Error               0          2m
kube-system     rke-user-addon-deploy-job-4hfx7           0/1       Error               0          1m
kube-system     rke-user-addon-deploy-job-578bm           0/1       Error               0          2m
kube-system     rke-user-addon-deploy-job-5bbvb           0/1       Error               0          1m
kube-system     rke-user-addon-deploy-job-5czws           0/1       Error               0          2m
.
.
.
kube-system     rke-user-addon-deploy-job-6cqpw           0/1       Error               0          2m
kube-system     rke-user-addon-deploy-job-6rzxd           0/1       Error               0          3m
kube-system     rke-user-addon-deploy-job-6z6qv           0/1       Error               0          14s
``` 
The job keeps creating pods and trying to execute and doesn't respect the `backoffLimit`. When we try to update a broken user addon in this case, it can take Kubernetes sometime to delete all the created pods, which, depending on how many pods are there, can trigger the timeout seen in https://github.com/rancher/rke/issues/718#issuecomment-401430333. Waiting till the number of created pods is > 200 seems to be enough to hit the timeout. 

If you re-try too quickly before enough pods are created, or you use the latest RC which has k8s `v1.10.5`, you will not be able to reproduce this timeout. The `backoffLimit` bug is fixed in `v1.10.5`.

**Comment by soumyalj** [comment]
The issue was reproduced only with K8s v1.10.3 when the addon_job_timeout is explicitly set low 10s or lower.[The default addon_job_timeout is 30s] As per the fix, the below error message is displayed to adjust the timeout value as below :
```
INFO[0069] [addons] Successfully Saved addon to Kubernetes ConfigMap: rke-user-addon 
INFO[0069] [addons] Executing deploy job..              
DEBU[0070] [k8s] replacing job rke-user-addon-deploy-job..  
ERRO[0091] [k8s] Job [rke-user-addon-deploy-job] deletion timedout. Consider increasing addon_job_timout value 
WARN[0091] Failed to deploy addon execute job [rke-user-includes-addons]: [k8s] Job [rke-user-addon-deploy-job] deletion timedout. Consider increasing addon_job_timout value 
INFO[0091] Finished building Kubernetes cluster successfully 
```
The issue is not reproduced with K8s v1.10.5 using rke version v0.1.8-rc11
